rust/src/nasl/builtin/network/udp.rs

```diff
@@ -0,0 +1,115 @@
+use std::{
+    io::{self, Read, Write},
+    net::{IpAddr, SocketAddr, UdpSocket},
+    os::fd::AsRawFd,
+    time::Duration,
+};
+
+use super::{mtu, network_utils::bind_local_socket};
+
+pub struct UdpConnection {
+    socket: UdpSocket,
+    buffer: Vec<u8>,
+    flags: Option<i32>,
+}
+
+const NUM_TIMES_TO_RESEND: usize = 5;
+
+impl Read for UdpConnection {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        for i in 0..NUM_TIMES_TO_RESEND {
+            let result = self.socket.recv_from(buf);
+            match result {
+                Ok((size, origin)) => {
+                    if self.socket.peer_addr()? == origin {
+                        return Ok(size);
+                    }
+                }
+                Err(e) if e.kind() == io::ErrorKind::TimedOut && i != NUM_TIMES_TO_RESEND - 1 => {
+                    self.socket.send(&self.buffer)?;
+                }
+                Err(e) => return Err(e),
+            };
+        }
+        unreachable!()
+    }
+}
+
+impl Write for UdpConnection {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let mtu = mtu(self.socket.peer_addr()?.ip());
+        if buf.len() < mtu {
```

Comment:
nit: I think guardian instruction are easier to read when the return early, so instead of writing it like this I would propose to switch the order:
```
if buf.len() > mtu {
  return Err(io::Error::new(...);
}
result = ...
Ok(result as usize)
```
That way a reader can immediately skip the guardian check without having to keep the branch in mind and concentrate on the result generation.